src/how-tos/mqtt/mqtt.kafka.broker.md

```diff
@@ -0,0 +1,201 @@
+
+Export these env variables or overwrite them with your remote Kafka if you skipped the local setup.
+
+```output:no-line-numbers
+export KAFKA_HOST=host.docker.internal
+export KAFKA_PORT=29092
+```
+
+### Bootstrap Kafka
+
+Create these topics in the Kafka environment.
+
+```bash:no-line-numbers
+/bin/kafka-topics.sh --bootstrap-server $KAFKA_HOST:$KAFKA_PORT --create --if-not-exists --topic mqtt-sessions
+/bin/kafka-topics.sh --bootstrap-server $KAFKA_HOST:$KAFKA_PORT --create --if-not-exists --topic mqtt-messages --config cleanup.policy=compact
+/bin/kafka-topics.sh --bootstrap-server $KAFKA_HOST:$KAFKA_PORT --create --if-not-exists --topic mqtt-retained --config cleanup.policy=compact
+```
+
+### Create your config
+
+Create a new file called `zilla.yaml` and append the below yaml to it.
+
+### Entrypoint
+
+This will configure Zilla for accepting all of the `mqtt` traffic.
+
+```yaml{14-16}
+<!-- @include: ./mqtt_kafka_broker_zilla.yaml{-20} -->
+```
+
+::: right
+[More on binding-tcp](../../reference/config/bindings/binding-tcp.md)
+[More on binding-mqtt](../../reference/config/bindings/binding-mqtt.md)
+:::
+
+### Service definition
+
+The service definitions will define how the clients using this service will interact with Kafka through Zilla.
+
+```yaml{7-9}
+<!-- @include: ./mqtt_kafka_broker_zilla.yaml{22-31} -->
+```
+
+::: right
+[More on binding-mqtt-kafka](../../reference/config/bindings/binding-mqtt-kafka.md)
+[More on topic data](../../concepts/kafka-proxies/mqtt-proxy.html#step-2-pub-sub-message-reflect-with-kafka)
+:::
+
+### Add a Kafka sync layer
+
+The Zilla [cache_client and cache_server](../../reference/config/bindings/binding-kafka.html#kind) helps manage the smooth data transfer between the service definition and Kafka.
+
+```yaml{11-12}
+<!-- @include: ./mqtt_kafka_broker_zilla.yaml{33-46} -->
+```
+
+::: right
+[More on binding-kafka cache](../../reference/config/bindings/binding-kafka.md#cache-behavior)
+:::
+
+### Point to a Running Kafka instance
+
+This will define the location and connection for Zilla to communicate with Kafka.
+
+```yaml{10-11}
+<!-- @include: ./mqtt_kafka_broker_zilla.yaml{47-} -->
+```
+
+::: details Full zilla.yaml
+
+```yaml
+<!-- @include: ./mqtt_kafka_broker_zilla.yaml -->
+```
+
+:::
+
+::: right
+[More on binding-kafka client](../../reference/config/bindings/binding-kafka.md#client-behavior)
+:::
+
+### Start Zilla
+
+With your `zilla.yaml` config, follow the [Zilla install instructions](../install.md) using your method of choice. Set the necessary Kafka environment variables.
+
+::: code-tabs#bash
+
+@tab Docker
+
+```bash:no-line-numbers
+--env KAFKA_HOST="$KAFKA_HOST" --env KAFKA_PORT="$KAFKA_PORT"
+```
+
+@tab Helm
+
+```bash:no-line-numbers
+--set extraEnv[1].value="$KAFKA_HOST",extraEnv [2].value="$KAFKA_PORT" \
```

Comment:
Space needs to move for `extraEnv[2]` ?

Reply:
Btw, is there a reason why we cannot do this by name instead of index? Seems a bit brittle and error prone.

Reply:
Yes it is, this is how to overwrite `value.yaml` list fields that I found. The `extraEnv` implementation is a 1:1 representation with the k8s deployment yaml env vars definition. 

Some charts allow for a config map to define more env vars or others have another param in the values yaml that expects simple key:value pair env vars. Here is an example from [grafana](https://github.com/helm/charts/blob/390ee6614b4bb0da96b397f9b382e3ddca9f59a4/stable/grafana/templates/_pod.tpl#L54C1-L54C1) We could add something like `extraEnvMap` or just use the `env` field like they do.

I think Atilla and I spoke about this and he was going to look more into it, then consult you.

Reply:
Idk if that answered the question but, the only reason why this isn't brittle is that I am overwriting the `values.yaml` that I created to deploy the chart. But definitely not ideal and since Zilla will need to leverage env vars to customize the zilla.yaml we should pick a mechanism we like and use that in the examples. Then the more ways we support users to define their own env vars the better.

Reply:
You rubber ducked me and I am realizing I need to do this inline env var overwrite in the `startup.sh` script but a person who is using this locally only needs to edit the local `values.yaml` file to point to their external Kafka. This is still a great chat to follow up on!